### PR TITLE
Support block toolbar

### PR DESCRIPTION
**Pull request by qinluhe**
### Feature Preview

- [x] Support `Add block below`
image
image

- [x] Support `Block Menu`
image
image

- [x] Support slash menu
image

---

#### PR Checklist

- [ ] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

**Comment by LucasXu0** [comment]
I think the logic for the `+` button should be:

1. if the current block is text block and it's empty, show the slash menu above it.
2. if not, insert a empty text block under the current block and then show the slash menu.

**Comment by qinluhe** [comment]
> I think the logic for the `+` button should be:
> 
> 1. if the current block is text block and it's empty, show the slash menu above it.
> 2. if not, insert a empty text block under the current block and then show the slash menu.

I haven't started writing the logic of slash yet, in the next PR

File: frontend/appflowy_tauri/src/appflowy_app/components/document/BlockSideToolbar/BlockMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import { List } from '@mui/material';
+import { ContentCopy, Delete } from '@mui/icons-material';
+import MenuItem from './MenuItem';
+import { useBlockMenu } from '$app/components/document/BlockSideToolbar/BlockMenu.hooks';
+import BlockMenuTurnInto from '$app/components/document/BlockSideToolbar/BlockMenuTurnInto';
+
+function BlockMenu({ id, onClose }: { id: string; onClose: () => void }) {
+  const { handleDelete, handleDuplicate } = useBlockMenu(id);
+
+  const [turnIntoPup, setTurnIntoPup] = React.useState<boolean>(false);
```
Comment (LucasXu0) [inline frontend/appflowy_tauri/src/appflowy_app/components/document/BlockSideToolbar/BlockMenu.tsx:11]: remove the `React`?

File: frontend/appflowy_tauri/src/appflowy_app/components/document/BlockSideToolbar/MenuItem.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  extra,
+  onHover,
+}: {
+  title: string;
+  icon: React.ReactNode;
+  onClick?: () => void;
+  extra?: React.ReactNode;
+  onHover?: (isHovered: boolean, event: React.MouseEvent<HTMLDivElement>) => void;
+}) {
+  return (
```
Comment (LucasXu0) [inline frontend/appflowy_tauri/src/appflowy_app/components/document/BlockSideToolbar/MenuItem.tsx:17]: Would it be better to wrap the parameters as a type or an interface?

I found a reference link.
https://www.technicalfeeder.com/2022/08/typescript-make-the-function-parameter-more-maintainable-by-using-interface/

Comment (qinluhe) [reply frontend/appflowy_tauri/src/appflowy_app/components/document/BlockSideToolbar/MenuItem.tsx:17]: In fact, it is the same. My current way of writing is to encapsulate it into an interface

**Review by LucasXu0 — APPROVED** [review]

**Pull request merged**